AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/UserOperations/AWSAuthResendSignUpCodeOperation.swift

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSPluginsCore
+import ClientRuntime
+import AWSCognitoIdentityProvider
+
+public class AWSAuthResendSignUpCodeOperation:
+    AmplifyOperation< AuthResendSignUpCodeRequest, AuthCodeDeliveryDetails, AuthError>, AuthResendSignUpCodeOperation {
+    
+    typealias CognitoUserPoolFactory = () throws -> CognitoUserPoolBehavior
+    private let userPoolFactory: CognitoUserPoolFactory
+    private var authConfiguration: AuthConfiguration
+    
+    init(_ request: AuthResendSignUpCodeRequest,
+         userPoolFactory: @escaping CognitoUserPoolFactory,
+         authConfiguration: AuthConfiguration,
+         resultListener: ResultListener?)
+    {
+        self.userPoolFactory = userPoolFactory
+        self.authConfiguration = authConfiguration
+        super.init(categoryType: .auth,
+                   eventName: HubPayload.EventName.Auth.resendSignUpCodeAPI,
+                   request: request,
+                   resultListener: resultListener)
+    }
+    
+    override public func main() {
+        if isCancelled {
+            finish()
+            return
+        }
+        
+        Task.init { [weak self] in
+            await self?.resendSignUpCode()
+        }
+    }
+    
+    func resendSignUpCode() async {
+        do {
+            let userPoolService = try userPoolFactory()
+            let clientMetaData = (request.options.pluginOptions
+                                  as? AWSResendSignUpCodeOptions)?.metadata ?? [:]
+            
+            let userPoolConfigurationData : UserPoolConfigurationData
+            switch authConfiguration {
+            case .userPools(let data):
+                userPoolConfigurationData = data
+            case .userPoolsAndIdentityPools(let data, _):
+                userPoolConfigurationData = data
+            case .identityPools:
+                let error = AuthError.configuration("UserPool configuration is invalid", AuthPluginErrorConstants.configurationError)
```

Comment:
Instead of this error message, should we say something like "UserPool configuration is missing"? Can you check how it is implemented in main branch, and use the same error here.

Reply:
Checked [here](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Operations/AWSAuthResendSignUpCodeOperation.swift) - there is no special handling of configuration error as I think `clientId` was not required for the SDK call. Updating the message to as suggested above